lib/railway_ipc/consumed_message_context.ex

```diff
@@ -0,0 +1,93 @@
+defmodule RailwayIpc.ConsumedMessageContext do
```

Comment:
Maybe this is the `MessageConsumption` Context (`Context` doesn't have to be explicit). All business logic about consuming messages is here. `MessageConsumption.process/4` seems ok. Not a big deal, but putting these under their business purpose as opposed to their entity seems like a good move.